src/controllers/filtercontroller.cpp

```diff
@@ -178,6 +187,9 @@ void FilterController::setCurrentFilter(int attachedIndex, bool isNew)
 
     emit currentFilterChanged(filter, meta, m_currentFilterIndex);
     m_currentFilter.reset(filter);
+    if (filter) {
+        filter->startUndoTracking(this);
```

Comment:
I just tried this, and it creates a problem because filter property setters need to be ignored when you add a new filter. With this change, you get a Change ___ filter command in history before the Add ____ filter command. We need to control when to start tracking changes to properties, which is why this is carefully placed here.